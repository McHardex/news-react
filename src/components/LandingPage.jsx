import React, { Component } from 'react'
import '../assets/stylesheets/landingPage.css'
import {Link} from 'react-router'
import SignUp from '../components/SignUp'

export class LandingPage extends Component {
  constructor(props) {
    super(props);
    this.state = { isOpen: false }

    this.openComponent = this.openComponent.bind(this)
    this.closeComponent = this.closeComponent.bind(this)
  }

  openComponent = (event) => {
    event.preventDefault();

    this.setState({ isOpen: true})
  }

  closeComponent = () => {
    this.setState({ isOpen: false})
  }

  render() {
    return (
      <div className='container'>
        <header>
          <div className='logo'>Mchardex::</div>
          <div className='getstarted'>get started <Link to='login' className='loginLink'>log in</Link></div>
        </header>
        <div className='section1'>
          <div className='sectionContent'>
            <p className='move'>Move Work Forward</p>
            <p className='create'>create as many articles as you can</p>
            <p className='create'>become a writer</p>
            <p className='move'>get new exiciting features</p>
            <button className='create-acc' onClick={this.openComponent}>create an account now</button>
          </div>
        </div>
        {this.state.isOpen && <SignUp closeComponent={this.closeComponent} /> }
        <footer>
          <span className='footerText'>become the face of <span className='foot-logo'> Mchardex:: </span> by writing exciting articles</span>
        </footer>
      </div>
    )
  }
}

export default LandingPage