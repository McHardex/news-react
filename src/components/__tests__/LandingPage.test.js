import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import LandingPage from '../LandingPage'

Enzyme.configure({ adapter: new Adapter()})

describe('The landing Page Component', () => {
  it('renders component without crashing', () => {
    const wrapper = shallow(<LandingPage />)
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.find('div').at(0).hasClass('container')).toBe(true);
    expect(wrapper.find('div').length).toEqual(5);
    expect(wrapper.find('p').length).toEqual(4);
    expect(wrapper.find('header').exists()).toBe(true);
    expect(wrapper.find('footer').exists()).toBe(true);
  });

  it('should call openComponent handler when button is clicked', () => {
    const wrapper = shallow(<LandingPage openComponent={jest.fn()}/>)
    wrapper.instance().openComponent = jest.fn()
    wrapper.instance().forceUpdate()
    wrapper.update()
    wrapper.find('button').simulate('click')
    expect(wrapper.instance().openComponent).toHaveBeenCalled()
  })

  it('should update the state when the button is clicked', () => {
    const wrapper = shallow(<LandingPage />)
    expect(wrapper.state('isOpen')).toBe(false)
    wrapper.find('button').simulate('click')
    expect(wrapper.state('isOpen')).toBe(true)
  })
});