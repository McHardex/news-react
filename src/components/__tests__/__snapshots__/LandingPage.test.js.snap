// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The display elements should not regress 1`] = `undefined`;
